tests/integration/test_examples.py

```diff
@@ -2,124 +2,83 @@
-    from within the notebook itself, so absolute paths can't be used.
-
-    When pytest is used to run the examples.py script (created directly from the
-    notebook), the script uses the actual cwd instead, which is (usually) the
-    project root dir. Therefore the examples.ipynb notebook and pytest-run
-    examples.py script both rely on the cwd being the same, but without
-    intervention it is different in each case.
-
-    Hence the test needs to set the cwd to the notebook's dir before running so
-    that the examples.py script uses the same cwd as the notebook.
+def examples_temp_data(tmp_path):
+    """Copy examples dir contents into temp dir for testing.
+
+    Any changes are discarded on fixture teardown.
+    :param tmp_path: temporary path fixture
+    :type tmp_path: Path
+    :return: temporary path containing examples files
+    :rtype: Path
     """
-    # Set up by storing cwd, which is usually the project root dir, then
-    # changing to examples/
-    cwd = Path.cwd()
-    os.chdir("examples")
-    yield
+    data_path = Path(__file__).parent.parent.parent / "examples"
+    copytree(data_path, tmp_path / "examples")
+    csv_json_path = (
+        Path(__file__).parent.parent.parent / "process/io/mfile_to_csv_vars.json"
+    )
+    copy(csv_json_path, tmp_path)
 
-    # Teardown by reverting cwd change
-    os.chdir(cwd)
+    # Return tmp_path/examples, now containing files copied from examples dir
+    return tmp_path / "examples"
 
 
-@pytest.fixture
-def delete_plot_procs(examples_as_cwd):
-    """Delete any plot_proc files produced by examples.ipynb.
-
-    :param examples_as_cwd: fixture to set examples dir as cwd
-    :type examples_as_cwd: None
-    """
-    yield
-    plot_proc_1 = Path("../examples/plot_proc_1.png")
-    plot_proc_2 = Path("../examples/plot_proc_2.png")
-    plot_proc_3 = Path("../examples/plot_proc_3.png")
-    plot_proc_1.unlink(missing_ok=True)
-    plot_proc_2.unlink(missing_ok=True)
-    plot_proc_3.unlink(missing_ok=True)
-
-
-def test_examples(delete_plot_procs):
-    """Run examples.ipynb and check no exceptions are raised.
+def test_examples(examples_temp_data):
+    """Run the examples.ipynb and check no exceptions are raised.
 
     examples.ipynb uses temp dirs to clean up any produced files itself.
-    """
-    with testbook("examples.ipynb", execute=True, timeout=600):
-        pass
-
-
-@pytest.fixture
-def scan_cleanup(examples_as_cwd):
-    """Delete any files produced by scan.ipynb.
+    :param examples_temp_data: temporary dir containing examples files
+    :type examples_temp_data: Path
 
-    :param examples_as_cwd: fixture to set examples dir as cwd
-    :type examples_as_cwd: None
     """
-    yield
-
-    # Teardown: delete produced files
-    scan_files = Path.cwd().glob("*scan_*")
-    for file in scan_files:
-        if "IN.DAT" not in file.name:
-            os.remove(file)
+    os.chdir(examples_temp_data)
```

Comment:
Can we do the change dir in the fixture (note this will remove the dir change from all of the tests in the file)